### PR TITLE
Tweaks metis xenos

### PR DESCRIPTION
**Pull request by Ketrai**
## About The Pull Request
- Randomizes the xenomorphs on the metis.
- Changes the hivebots to be randomly carp, hivebots, or spiders instead.
- One guaranteed borer spawn, 35% chance for a second one to spawn.
- Neutered borer is now their own subtype, so no need to varedit them in anymore.
- Neutered borer is now named chemslug, to seem less intimidating and more as a xenobio pet, which is apparently the intent by the original mapmaker.
- Adds element for mob appearance, and mob name upon login.
- Allows neutered borers to pick their appearance as little subspecies of brainslugs. Be mysterious! Scary! Cute! The world is your oyster.
- Allows neutered borers to pick their own name, so they can have a pet name, as dictated by their imagined, former owner(s). This does not change their true name.

## Why It's Good For The Game
Makes the metis a bit more interesting for replay value. Instead of just fighting the same creatures each time, they can now be varied!

Nurse spiders can't spawn, would rather not have a mass outbreak with the amount of monkeys around.

Borers are kinda antag-y by nature. And whilst sterile borers aren't meant as antags, they still look freaky, have a freaky name, etc. This might confuse both players who pick borer, and those who interact with sterile borers. Now there's less reason to immediately murder them.

image

## Changelog
:cl:
tweak: Metis is now more randomized!
tweak: Neutered borers are now less intimidating.
/:cl:

### CODE REVIEW

**Comment by Latentish** [comment]
I don't really feel that spiders or carp are thematically appropriate, and I feel like adding a spawn chance to the borer removes it's purpose. Part of the metis's appeal is the borer spawn, so having a chance for that not to happen detracts from the appeal of the ship. The Borer is also very good for helping to deal with the hivebots and xenos on the ship. Also a second borer is super unnecessary, as the first borer can just make a second one.
Adding the xeno queen to the spawn chance is also probably not a good idea as well, given that it means they might have a chance to multiply.

**Comment by PrefabQuasar** [comment]
> I don't really feel that spiders or carp are thematically appropriate, and I feel like adding a spawn chance to the borer removes it's purpose. Part of the metis's appeal is the borer spawn, so having a chance for that not to happen detracts from the appeal of the ship. The Borer is also very good for helping to deal with the hivebots and xenos on the ship. Also a second borer is super unnecessary, as the first borer can just make a second one. Adding the xeno queen to the spawn chance is also probably not a good idea as well, given that it means they might have a chance to multiply.

the borer is NOT meant to be able to make more borers, the only reason its present at all was because I had been informed that it had been fixed and they would be properly sterile now

**Comment by Latentish** [comment]
> the borer is NOT meant to be able to make more borers, the only reason its present at all was because I had been informed that it had been fixed and they would be properly sterile now

Ah in that case, they should get fixed altogether, and I can understand a chance for a second one to spawn. But please no chance for it to not spawn at all.

**Comment by Ketrai** [comment]
> I don't really feel that spiders or carp are thematically appropriate, and I feel like adding a spawn chance to the borer removes it's purpose. Part of the metis's appeal is the borer spawn, so having a chance for that not to happen detracts from the appeal of the ship. The Borer is also very good for helping to deal with the hivebots and xenos on the ship. Also a second borer is super unnecessary, as the first borer can just make a second one. Adding the xeno queen to the spawn chance is also probably not a good idea as well, given that it means they might have a chance to multiply.

As far as I can tell, the xeno queens do not reproduce. I was waiting around for a while just to make sure, and they don't seem to spawn eggs. If anything, when AI-controlled, they seems to be mostly just tankier basic xenos, doing less melee damage than hunters.

The borer being a guaranteed spawn just feels a bit gimmicky. And the chance for one to spawn is still really, really high. Alternatively, if others also want it to still be guaranteed, I can just give it a couple random spawn locations, and when it spawns once it won't trigger the other locations. So at least you won't always find one in the xenobiology section, but other areas instead.

Dealing with the xenos isn't really too hard, and you can get a drone spawn to help with that as well. At least half the mobs tends to die to fighting other factions, or getting smacked with the laser in AI. Either way having a second player, whether it be crew, a borer, or a drone tends to help immensely. And all guns are easily accessible without running into danger.

As for carp/spiders not being thematic I'm not sure what you mean? As far as I can tell, it's a ship that's suffering a xeno outbreak thanks to the mistakes of the former before the rest of them wake up.

**Comment by PiperDoots** [comment]
> > I don't really feel that spiders or carp are thematically appropriate, and I feel like adding a spawn chance to the borer removes it's purpose. Part of the metis's appeal is the borer spawn, so having a chance for that not to happen detracts from the appeal of the ship. The Borer is also very good for helping to deal with the hivebots and xenos on the ship. Also a second borer is super unnecessary, as the first borer can just make a second one. Adding the xeno queen to the spawn chance is also probably not a good idea as well, given that it means they might have a chance to multiply.
> 
> As far as I can tell, the xeno queens do not reproduce. I was waiting around for a while just to make sure, and they don't seem to spawn eggs. If anything, when AI-controlled, they seems to be mostly just tankier basic xenos, doing less melee damage than hunters.
> 
> The borer being a guaranteed spawn just feels a bit gimmicky. And the chance for one to spawn is still really, really high. Alternatively, if others also want it to still be guaranteed, I can just give it a couple random spawn locations, and when it spawns once it won't trigger the other locations. So at least you won't always find one in the xenobiology section, but other areas instead.
> 
> Dealing with the xenos isn't really too hard, and you can get a drone spawn to help with that as well. At least half the mobs tends to die to fighting other factions, or getting smacked with the laser in AI. Either way having a second player, whether it be crew, a borer, or a drone tends to help immensely. And all guns are easily accessible without running into danger.
> 
> As for carp/spiders not being thematic I'm not sure what you mean? As far as I can tell, it's a ship that's suffering a xeno outbreak thanks to the mistakes of the former before the rest of them wake up.

Personally think the borer is a pretty integral part of the ship, and one of the few ghost roles we have. Do like the idea of giving it a random spawn location though.

**Comment by Latentish** [comment]
> As far as I can tell, the xeno queens do not reproduce. I was waiting around for a while just to make sure, and they don't seem to spawn eggs. If anything, when AI-controlled, they seems to be mostly just tankier basic xenos, doing less melee damage than hunters.

- Oh ok my mistake then. 

> The borer being a guaranteed spawn just feels a bit gimmicky. And the chance for one to spawn is still really, really high. Alternatively, if others also want it to still be guaranteed, I can just give it a couple random spawn locations, and when it spawns once it won't trigger the other locations. So at least you won't always find one in the xenobiology section, but other areas instead.

- I have to heavily encourage you to go with the alternative, as it's a huge part of the charm. I would largely see people simply ahelping for a borer if it didn't spawn because (but this is the borer ship). Though as Prefab said, them reproducing was never meant to be a thing so I can see having a second possible spawn as a positive change.

> As for carp/spiders not being thematic I'm not sure what you mean? As far as I can tell, it's a ship that's suffering a xeno outbreak thanks to the mistakes of the former before the rest of them wake up.

- I meant that carp and spiders are much more mundane creature. Hivebots are this weird alien robot thing, and xenos are murder aliens. Maybe instead of spiders and carp, do something like otherworld creatures. Something more alien than fish and bugs.

**Comment by PrefabQuasar** [comment]
honestly uh
when i put that borer spawn in there the intent was that it was a known thing the crew had around, like a ship pet, which is why theres sunglasses, a radio, and a shot glass on the floor next to it

**Comment by Ketrai** [comment]
> * I meant that carp and spiders are much more mundane creature. Hivebots are this weird alien robot thing, and xenos are murder aliens. Maybe instead of spiders and carp, do something like otherworld creatures. Something more alien than fish and bugs.

Oh yeah, honestly I agree. But there's not that many creatures I want to risk putting in considering they quickly get to be a lot deadlier than hivebots, which are really squishy to begin with. I can always sprite some more alien, weak creatures later. But It would be out of the scope for this PR.

**Comment by Ketrai** [comment]
> honestly uh when i put that borer spawn in there the intent was that it was a known thing the crew had around, like a ship pet, which is why theres sunglasses, a radio, and a shot glass on the floor next to it

I was wondering why the sunglasses were there. The area kinda screams. "Dangerous Alien" considering it's in the bloodied part of the ship, and in one of the xenobio containment cells, and on top there's a corpse those items could've belonged to right next to it, it sends confusing messages. They could make more sense either in the RD's office as another pet like lamar, or the cryo room?

EDIT : Also I know borers don't typically have names, but can always spawn it in as a "pet borer" and then give the player a small prompt allowing them to update the name of their borer when they take control. With the flavor prompt for possessing the borer stating you should pick your name as a pet of the crew? That would make the intended purpose a lot less confusing for sure, as right now any old player might assume they're evil or otherwise.

File: code/game/objects/effects/spawners/mobspawner.dm [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+				/obj/effect/spawner/lootdrop/carpspawner
+			)
+		if(3)
+			loot = list(
+				/obj/effect/spawner/lootdrop/spiderspawner
+			)
+	. = ..()
+
+/obj/effect/spawner/lootdrop/xenospawner
+	name = "Improper xenomorph spawner"
```
Comment (Fikou) [inline code/game/objects/effects/spawners/mobspawner.dm]: what are you doing with the names?

File: code/game/objects/effects/spawners/mobspawner.dm [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+			)
+		if(2)
+			loot = list(
+				/obj/effect/spawner/lootdrop/carpspawner
+			)
+		if(3)
+			loot = list(
+				/obj/effect/spawner/lootdrop/spiderspawner
+			)
+	. = ..()
```
Comment (Fikou) [inline code/game/objects/effects/spawners/mobspawner.dm]: return ..() at the end of procs

File: code/game/objects/effects/spawners/mobspawner.dm [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+		/mob/living/simple_animal/hostile/alien/drone = 0.2,
+		/mob/living/simple_animal/hostile/alien/sentinel = 0.3,
+		/obj/effect/spawner/lootdrop/xenoqueenspawner = 0.1
+	)
+
+/obj/effect/spawner/lootdrop/xenoqueenspawner
+	name = "Improper xenomorph queen spawner"
+	loot = list(
+		/mob/living/simple_animal/hostile/alien/queen = 0.999, //regular queen mob isn't actually that strong
+		/mob/living/simple_animal/hostile/alien/queen/large = 0.001
```
Comment (Fikou) [inline code/game/objects/effects/spawners/mobspawner.dm]: whats the point here, 0.01% chance for a simplemob thats not even like, very strong

File: code/game/objects/effects/spawners/mobspawner.dm [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+		/mob/living/simple_animal/hostile/poison/giant_spider,
+		/mob/living/simple_animal/hostile/poison/giant_spider/hunter,
+		/mob/living/simple_animal/hostile/poison/giant_spider/hunter/viper,
+	)
+
+/obj/effect/spawner/lootdrop/carpspawner
+	name = "Improper carp spawner"
+
+	loot = list(
+		/mob/living/simple_animal/hostile/carp,
```
Comment (Fikou) [inline code/game/objects/effects/spawners/mobspawner.dm]: why are you having a spawner for one object

**Comment by Sylphily** [comment]
Having the borer be only a chance is dumb and they can't replicate last I checked

**Comment by Ketrai** [comment]
> Having the borer be only a chance is dumb and they can't replicate last I checked

Don't worry, there will be a guaranteed borer again, with a chance for a second. I'm just working to expand this PR so you can also pick a borer sprite and name, to look less intimidating, or more intimidating if that's your style.

File: code/game/objects/effects/spawners/mobspawner.dm (hunk 3 above)
Comment (Ketrai) [reply code/game/objects/effects/spawners/mobspawner.dm]: I'll pop it to 20%

File: code/game/objects/effects/spawners/mobspawner.dm (hunk 4 above)
Comment (Ketrai) [reply code/game/objects/effects/spawners/mobspawner.dm]: True

File: code/game/objects/effects/spawners/mobspawner.dm (hunk 1 above)
Comment (Ketrai) [reply code/game/objects/effects/spawners/mobspawner.dm]: Sometimes, I don't even know what I'm doing.

**Comment by Ketrai** [comment]
Main description updated.

**Comment by Latentish** [comment]
To be honest, I vastly prefer the old sprites. I don't really think a new sprite should totally change their look. The art style also just doesn't feel like it fits with the rest of the server's artstyle. I'd recommend talking to triplezeta or our other host of good spriters for help with a direction for them.

Also, you're using the whitesands folder which is completely deprecated, so you'll need to move those files out to their proper place.
Also please do extensively test this, as it has 2 failing checks, and is actively requesting review.

Edit: I also realized that if you're going to make the borer change spawn location ship to ship, make sure the items it's supposed to spawn next to also move with it. Doesn't make sense for there to just be sunglasses, a radio, and a shot glass in xenobio without the borer that owns them.

I also feel I should reiterate, please change the selection of mobs you're putting in the spawner to something more appropriate than spiders and carp, if you haven't already. It doesn't really make sense to add a spawner that adds non-thematic spawns.

**Comment by Ketrai** [comment]
> To be honest, I vastly prefer the old sprites. I don't really think a new sprite should totally change their look. The art style also just doesn't feel like it fits with the rest of the server's artstyle. I'd recommend talking to triplezeta or our other host of good spriters for help with a direction for them.

Isn't the style just an unholy mix of /TG/, classic SS13, eris, shiptest sprites? Without sprite consistency I can't really keep a "style" in mind, sorry. I've tried these and they don't look weird ingame. I've even made sure to keep the colors from going all too bright, so they don't stand out. 

> Also, you're using the whitesands folder which is completely deprecated, so you'll need to move those files out to their proper place. Also please do extensively test this, as it has 2 failing checks, and is actively requesting review.

The linter check? I'm not sure what it's upset about. It says indentation, but I'm not seeing any weird indentation so it might've derped. (BY THE BY, Linters don't mean anything is broken, it's mostly just about keeping consistant formatting and avoiding potential breaks due to byond goofery) Earlier it was just poking at the lack of a new line, which I've added to every component. I was told by triplezeta not to worry about the whitesands icon file. Touching any of those automatically triggers a review from triplezeta. But if you're really concerned, I can just move the icon file, it's not a big deal. 

I also don't know what mapbot2's problem is, as it seems to just crash for all map PRs currently. It's not even getting executed.

> Edit: I also realized that if you're going to make the borer change spawn location ship to ship, make sure the items it's supposed to spawn next to also move with it. Doesn't make sense for there to just be sunglasses, a radio, and a shot glass in xenobio without the borer that owns them.

I'm not moving them anymore anyways. I just want for them to be less intimidating at this point. 

> I also feel I should reiterate, please change the selection of mobs you're putting in the spawner to something more appropriate than spiders and carp, if you haven't already. It doesn't really make sense to add a spawner that adds non-thematic spawns.

Then suggest something? Many mobs are pretty deadly, I don't want to use anything too awful.

**Comment by Ketrai** [comment]
image

**Comment by Latentish** [comment]
> Isn't the style just an unholy mix of /TG/, classic SS13, eris, shiptest sprites? Without sprite consistency I can't really keep a "style" in mind, sorry. I've tried these and they don't look weird ingame. I've even made sure to keep the colors from going all too bright, so they don't stand out.

-I don't particularly feel so. We have a pretty distinguishable emerging artstyle from what  different codebases are pulled from, as well as the insane work done by triplezeta and retlaw. Some examples being:
https://github.com/shiptest-ss13/Shiptest/pull/1122
https://github.com/shiptest-ss13/Shiptest/pull/1119
https://github.com/shiptest-ss13/Shiptest/pull/1115
https://github.com/shiptest-ss13/Shiptest/pull/1114
While these aren't mobs, they do have a pretty consistent look to them imo. But my bigger concern is making them not look like borers. Borers have a pretty recognizable shape and palette. They're green and they got the funny feelers out front. I'm just reticent about that being outright gone in any form, especially on the Metis.


> The linter check? I'm not sure what it's upset about. It says indentation, but I'm not seeing any weird indentation so it might've derped. (BY THE BY, Linters don't mean anything is broken, it's mostly just about keeping consistant formatting and avoiding potential breaks due to byond goofery) Earlier it was just poking at the lack of a new line, which I've added to every component. I was told by triplezeta not to worry about the whitesands icon file. Touching any of those automatically triggers a review from triplezeta. But if you're really concerned, I can just move the icon file, it's not a big deal.

-A big deal or not, it's still a pretty good practice to keep consistent formatting, and not add into the whitesands folder, since there have been several recent PR's moving things out of them. If TripleZeta said otherwise, I'm no one to argue, but it wasn't meant as some kind of attack. It's good that you got it fixed. As for the mapdiffbot, yeah it's been going off constantly, something something docking_port/mobile.
  

> Then suggest something? Many mobs are pretty deadly, I don't want to use anything too awful.

- Skeletons, migos, blankbodies, creature (hostile/netherworld), faithless, most constructs, and most of the clowns (though some people might object).

**Comment by Ketrai** [comment]
> -I don't particularly feel so. We have a pretty distinguishable emerging artstyle from what different codebases are pulled from, as well as the insane work done by triplezeta and retlaw. Some examples being: #1122 #1119 #1115 #1114 While these aren't mobs, they do have a pretty consistent look to them imo. But my bigger concern is making them not look like borers. Borers have a pretty recognizable shape and palette. They're green and they got the funny feelers out front. I'm just reticent about that being outright gone in any form, especially on the Metis.

Well if you care to look at the pr's changes in detail. Players are free to choose their borer appearance. The old sprite is still in there as an option for that more telltale look. I just don't like the way 1-thick lines extending out of a sprite look. They feel busy. Can just leave those details to the imagination. Other than that, there's no specific style-guide, and the sprites themselves, in my eyes, are of good enough quality. It'd be pretty sad if I couldn't manage with my 2+ years of experience as a pixelartist. This feels more like you're trying to gatekeep me from contributing.

> * Skeletons, migos, blankbodies, creature (hostile/netherworld), faithless, most constructs, and most of the clowns (though some people might object).

Not sure about skeletons. They're a bit of a weird one to throw in there too. Faithless I'd have to test. Migos and the creature are both rather dangerous though. Especially since they're really fast and just A-line it to the player. I'd rather see what actual maintainers have to say about this one.

> -A big deal or not, it's still a pretty good practice to keep consistent formatting, and not add into the whitesands folder, since there have been several recent PR's moving things out of them. If TripleZeta said otherwise, I'm no one to argue, but it wasn't meant as some kind of attack. It's good that you got it fixed. As for the mapdiffbot, yeah it's been going off constantly, something something docking_port/mobile.

It's less that, more that you specifically, are making a big deal out of it. It's just the way you come at me, with a complete feeling of untrust that I feel forced into a defensive position. Especially with phrases like "Please extensively test this" As if I've been making badly tested prs for weeks already here.

**Comment by Ketrai** [comment]
Also talking about sprites, isn't that migo sprite a straight rip from a tileset in Cataclysm, Dark Days Ahead >:C
(I'd rather not use mobs with potentially ripped sprites)
image
image

**Comment by triplezeta** [comment]
@Apogee-dev @SweetBlueSylveon

**Review by triplezeta — DISMISSED** [review]
Pretty neat PR overall, I believe that the colorful and coral borertypes could use more work in regards to shading however.

**Comment by Ketrai** [comment]
> Pretty neat PR overall, I believe that the colorful and coral borertypes could use more work in regards to shading however.

I'm happy to, but only with actual suggestions towards the shading. There's not a lot of pixels to work with here.

**Comment by triplezeta** [comment]
i was going to get to that, just wanted to leave that there before i forgot

File: code/modules/antagonists/borer/borer.dm [hunk 5]
```diff
@@ -122,6 +122,16 @@ GLOBAL_VAR_INIT(total_borer_hosts_needed, 3)
 	var/is_team_borer = TRUE
 	var/borer_alert = "Become a cortical borer? (Warning, You can no longer be cloned!)"
 
+/mob/living/simple_animal/borer/sterile
+	name = "neutered chemslug"
+	is_team_borer = FALSE
+	borer_alert = "Become a neutered cortical borer? (Warning, You can no longer be cloned!)"
+
+/mob/living/simple_animal/borer/sterile/Initialize(mapload)
```
Comment (MarkSuckerberg) [inline code/modules/antagonists/borer/borer.dm]:
```suggestion
/mob/living/simple_animal/borer/sterile/ComponentInitialize()
```

**Review by MarkSuckerberg — CHANGES_REQUESTED** [review]
Other than this seems good codewise

**Comment by Ketrai** [comment]
I got new sprites for these. I'll apply them when I'm back from holidays

**Review by Fikou — APPROVED** [review]

**Comment by Ketrai** [comment]
image

**Comment by Ketrai** [comment]
Mods!

**Review by MarkSuckerberg — APPROVED** [review]
We win these!

**Pull request merged**